src/specializations/Triangle.jl

```diff
@@ -12,115 +12,29 @@
-        triangle::Meshes.Triangle,
-        rule::HAdaptiveCubature;
-        diff_method::DM = Analytical(),
-        FP::Type{T} = Float64
-) where {DM <: DifferentiationMethod, T <: AbstractFloat}
-    _guarantee_analytical(Meshes.Triangle, diff_method)
-
-    # Integrate the Barycentric triangle by transforming it into polar coordinates
-    #   with a modified radius
-    #     R = r ( sinφ + cosφ )
-    #   s.t. integration bounds become rectangular
-    #     R ∈ [0, 1] and φ ∈ [0, π/2]
-    function integrand(Rφ)
-        R, φ = Rφ
-        a, b = sincos(φ)
-        u = R * (1 - a / (a + b))
-        v = R * (1 - b / (a + b))
-        return f(triangle(u, v)) * R / (a + b)^2
-    end
-
-    # HCubature doesn't support functions that output Unitful Quantity types
-    # Establish the units that are output by f
-    testpoint_parametriccoord = _zeros(T, 2)
-    integrandunits = Unitful.unit.(integrand(testpoint_parametriccoord))
-    # Create a wrapper that returns only the value component in those units
-    uintegrand(ts) = Unitful.ustrip.(integrandunits, integrand(ts))
-    # Integrate only the unitless values
-    ∫ = HCubature.hcubature(uintegrand, _zeros(FP, 2), (T(1), T(π / 2)), rule.kwargs...)[1]
-
-    # Apply a linear domain-correction factor 0.5 ↦ area(triangle)
-    return 2 * Meshes.area(triangle) * integrandunits .* ∫
+        rule::IntegrationRule;
+        kwargs...
+)
+    # Generate a _ParametricGeometry whose parametric function domain spans [0,1]²
+    param_triangle = _ParametricGeometry(_parametric(triangle), 2)
+
+    # Integrate the _ParametricGeometry using the standard methods
+    return _integral(f, param_triangle, rule; kwargs...)
 end
 
-################################################################################
-#                               jacobian
-################################################################################
-
-_has_analytical(::Type{T}) where {T <: Meshes.Triangle} = true
-
 ################################################################################
 #                              Parametric
 ################################################################################
 
-function _parametric(triangle::Meshes.Triangle, t1, t2)
-    if (t1 < 0 || t1 > 1) || (t2 < 0 || t2 > 1)
-        msg = "triangle(t1, t2) is not defined for (t1, t2) outside [0, 1]^2."
-        throw(DomainError((t1, t2), msg))
-    end
-
-    # Form a line segment between sides
-    a = triangle(0, t2)
-    b = triangle(t2, 0)
-    segment = Meshes.Segment(a, b)
+# Map argument domain from [0, 1]² to Barycentric domain for (::Triangle)(t1, t2)
+function _parametric(triangle::Meshes.Triangle)
+    function f(t1, t2)
+        if any(Iterators.map(n -> (n < 0) || (n > 1), (t1, t2)))
+            msg = "triangle(t1, t2) is not defined for (t1, t2) outside [0, 1]²."
+            throw(DomainError((t1, t2), msg))
+        end
 
-    return segment(t1)
+        return triangle(t1 * t2, t2 - (t1 * t2))
```

Comment:
```suggestion
        t1t2 = t1 * t2
        return triangle(t1t2, t2 - t1t2)
```
I don't know how much we gain from this, but since this function is performance critical maybe saving one multiplication in each call gives us a small performance gain? We could at least try.